projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/WPCOM_REST_API_V2_Endpoint_Block_Editor_Assets_Test.php

```diff
@@ -130,4 +130,149 @@ public function test_get_items() {
 		$this->assertIsString( $data['styles'] );
 		$this->assertIsString( $data['scripts'] );
 	}
+
+	/**
+	 * Test that get_items returns the expected block types.
+	 */
+	public function test_get_items_returns_allowed_block_types() {
+		wp_set_current_user( self::factory()->user->create( array( 'role' => 'editor' ) ) );
+
+		$request  = new WP_REST_Request( Requests::GET, '/wpcom/v2/editor-assets' );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		// Verify allowed block types are present
+		$this->assertArrayHasKey( 'allowed_block_types', $data );
+		$this->assertIsArray( $data['allowed_block_types'] );
+
+		// Test core blocks are included
+		$core_blocks = array_filter(
+			$data['allowed_block_types'],
+			function ( $block_name ) {
+				return strpos( $block_name, 'core/' ) === 0;
+			}
+		);
+		$this->assertNotEmpty( $core_blocks );
+
+		// Test Jetpack blocks are included
+		$jetpack_blocks = array_filter(
+			$data['allowed_block_types'],
+			function ( $block_name ) {
+				return strpos( $block_name, 'jetpack/' ) === 0;
+			}
+		);
+		$this->assertNotEmpty( $jetpack_blocks );
+
+		// Test specific known blocks are present
+		$this->assertContains( 'jetpack/contact-form', $data['allowed_block_types'] );
+		$this->assertContains( 'jetpack/subscriptions', $data['allowed_block_types'] );
+	}
+
```

Comment:
It might be a good idea to add a test for blocks that are currently not in the ALLOWED_PLUGIN_BLOCKS list. So that folks remember to add them or intentionally add them to the do not include list.